src/server/datasource/ncbi.js

```diff
@@ -176,44 +165,87 @@ const mergeStrains = function(){
       return updateChunk().then( removeMergedDescendants );
     };
 
-    const registerUpdates = function( ancestors, descendants ){
-      ancestors.forEach( ( ancestor, i ) => {
-        let descendant = descendants[ i ];
-        let updates = updateMap.has( descendant.name ) ?
-          updateMap.get( descendant.name ) : { synonyms: [], ids: [], organisms: [] };
+    const registerUpdates = function( roots, mergeFromLists ){
+      roots.forEach( ( root, i ) => {
+        let mergeFromList = mergeFromLists[ i ];
+
+        if ( mergeFromList.length == 0 ) {
+          throw 'mergeFromList must not be empty!';
```

Comment:
It's usually better to `throw new Error(string)` rather than `throw string`.

Reply:
@maxkfranz I updated it.